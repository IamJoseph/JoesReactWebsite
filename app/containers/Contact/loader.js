/**
 * Asynchronously loads the components for Contact
 */

import { errorLoading, getAsyncInjectors } from 'utils/asyncInjectors';

export default (store) => {
  const { injectReducer, injectSagas } = getAsyncInjectors(store);
  return (cb) => {
    const importModules = Promise.all([
      import('containers/Contact/reducer'),
      import('containers/Contact/sagas'),
      import('containers/Contact'),
    ]);

    importModules.then(([reducer, sagas, component]) => {
      injectReducer('form', reducer.default);
      injectSagas(sagas.default);
      cb(component);
    });

    importModules.catch(errorLoading);
  };
};
