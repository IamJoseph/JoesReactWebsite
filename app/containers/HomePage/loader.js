/**
 * Asynchronously loads the components for HomePage
 */

import { errorLoading, getAsyncInjectors } from 'utils/asyncInjectors';

export default (store) => {
  const { injectReducer, injectSagas } = getAsyncInjectors(store);
  return (cb) => {
    const importModules = Promise.all([
      import('containers/HomePage/Astronaut/reducer'),
      import('containers/HomePage/Astronaut/sagas'),
      import('containers/HomePage'),
    ]);

    importModules.then(([reducer, sagas, component]) => {
      injectReducer('home', reducer.default);
      injectSagas(sagas.default);

      cb(component);
    });

    importModules.catch(errorLoading);
  };
};
