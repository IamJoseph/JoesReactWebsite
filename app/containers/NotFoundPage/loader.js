/**
 * Asynchronously loads the components for NotFoundPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/NotFoundPage')
    .then(cb)
    .catch(errorLoading);
};
