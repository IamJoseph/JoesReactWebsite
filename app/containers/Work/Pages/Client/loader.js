/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work/Pages/Client')
    .then(cb)
    .catch(errorLoading);
};
