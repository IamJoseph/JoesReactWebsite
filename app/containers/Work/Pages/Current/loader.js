/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work/Pages/Current')
    .then(cb)
    .catch(errorLoading);
};
