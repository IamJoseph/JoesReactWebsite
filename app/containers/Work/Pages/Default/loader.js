/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work/Pages/Default')
    .then(cb)
    .catch(errorLoading);
};
