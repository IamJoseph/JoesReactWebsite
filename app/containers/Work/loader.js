/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work')
    .then(cb)
    .catch(errorLoading);
};
