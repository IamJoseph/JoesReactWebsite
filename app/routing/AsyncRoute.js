/**
 * A helper component that renders a Route that will load a component when the Route is
 * rendered. This allows for asynchronous loading of routes, enabling code-splitting.
 *
 * Inspired by https://reacttraining.com/react-router/web/guides/code-splitting
 */

import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';

/**
 * A wrapper component that will lazily render a component after it has been loaded.
 */
export class Bundle extends Component {
  state = {
    // short for "module" but that's a keyword in js, so "mod"
    mod: null,
  };
  componentWillMount() {
    this.load(this.props);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.load !== this.props.load) {
      this.load(nextProps);
    }
  }

  // mod.default function that returns react component
  load(props) {
    this.setState({
      mod: null,
    });
    props.load((mod) => {
      this.setState({
        // handle both es imports and cjs
        mod: mod.default ? mod.default : mod,
      });
    });
  }

  render() {
    // eslint-disable-next-line no-unused-vars
    const { load, ...otherProps } = this.props;
    return this.state.mod && <this.state.mod {...otherProps} />;
  }
}

Bundle.propTypes = {
  load: PropTypes.func.isRequired,
};

// wrap <Route> and use this everywhere instead, then when
// sub routes are added to any route it'll work
const AsyncRoute = ({ path, load }) => (
  <Route
    path={path}
    render={(props) => (
      <Bundle load={load} {...props} />
    )}
  />
);


AsyncRoute.propTypes = {
  path: PropTypes.string.isRequired,
  load: PropTypes.func.isRequired,
};

export default AsyncRoute;
