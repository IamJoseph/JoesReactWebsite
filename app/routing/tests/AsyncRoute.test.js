import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import loader from 'containers/Work/loader';
import AsyncRoute from '../AsyncRoute';

describe('<AsyncRoute />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <AsyncRoute
        exact path="/work" load={loader()}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
