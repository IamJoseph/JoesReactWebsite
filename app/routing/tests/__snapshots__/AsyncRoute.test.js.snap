// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AsyncRoute /> should render correctly 1`] = `
<Route
  path="/work"
  render={[Function]}
/>
`;
